datasync/chaindatafetcher/kas/repository_transactions.go

```diff
@@ -0,0 +1,178 @@
+package kas
+
+import (
+	"fmt"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"strings"
+)
+
+// TxFilteringTypes filters types which are only stored in KAS database.
+var TxFilteringTypes = map[types.TxType]bool{
+	types.TxTypeLegacyTransaction: true,
+
+	types.TxTypeValueTransfer:                      true,
+	types.TxTypeFeeDelegatedValueTransfer:          true,
+	types.TxTypeFeeDelegatedValueTransferWithRatio: true,
+
+	types.TxTypeValueTransferMemo:                      true,
+	types.TxTypeFeeDelegatedValueTransferMemo:          true,
+	types.TxTypeFeeDelegatedValueTransferMemoWithRatio: true,
+
+	types.TxTypeSmartContractDeploy:                      true,
+	types.TxTypeFeeDelegatedSmartContractDeploy:          true,
+	types.TxTypeFeeDelegatedSmartContractDeployWithRatio: true,
+
+	types.TxTypeSmartContractExecution:                      true,
+	types.TxTypeFeeDelegatedSmartContractExecution:          true,
+	types.TxTypeFeeDelegatedSmartContractExecutionWithRatio: true,
+}
+
+// transformToTxs transforms a chain event to transaction list according to the KAS database scheme.
+func transformToTxs(event blockchain.ChainEvent) []*Tx {
+	var txs []*Tx
+	// TODO-ChainDataFetcher extract EOAs in order to expire the KAS cache
+	//expiredEOAs := make(map[common.Address]struct{})
+
+	block := event.Block
+	head := block.Header()
+	receipts := event.Receipts
+
+	for idx, rawTx := range block.Transactions() {
+		txId := head.Number.Int64()*maxTransactionCount*maxTransactionLogCount + int64(idx)*maxInternalTransactionCount
+
+		// from
+		var from common.Address
+		if rawTx.IsLegacyTransaction() {
+			signer := types.NewEIP155Signer(rawTx.ChainId())
+			from, _ = types.Sender(signer, rawTx)
+		} else {
+			from, _ = rawTx.From()
+		}
+
+		// to
+		to := rawTx.To()
+		if to == nil {
+			to = &common.Address{}
+		}
```

Comment:
Is it okay if `to` value is not updated to the generated contract address?

Reply:
Yes, we don't store the contract address in `klay_transfers` table.